### ISSUE TITLE
dply::left_join error in convert_glatos_to_att()

### ISSUE THREAD

**Issue by chrisholbrook** [ocean-tracking-network/glatos#251]
Error encountered with convert_glatos_to_att() when input detectionObj contains 6 million rows: 

>Error in `dplyr::left_join()` at glatos/R/util-convert_glatos_to_att.r:115:3:
>! This join would result in more rows than dplyr can handle.

```
# dtc_file is a GLATOS detection export csv file (from GLATOS data portal)
# rcv_file is a GLATOS receiver locations csv file (from GLATOS data portal)

detectionObj <- read_glatos_detections(dtc_file)
receiverObj <- read_glatos_receivers(rcv_file)
  

dtc_att <- convert_glatos_to_att(detectionObj, 
                                 receiverObj, 
                                 crs = sf::st_crs(4326))
```

![image](https://github.com/user-attachments/assets/994d6d5a-161c-4d59-b7cb-184e08294bef)

**Comment by mhpob** [comment]
I don't believe this is a length thing at 6 million rows (though that error says it probably will be at 2.1 billion!): I created a toy dataset with 6m rows and it ran fine.

It should be that detectionObj dictates the maximum amount of rows, assuming there are fewer station records than detections or number of fish detected. We're seeing 400x the rows than the input dataset.

The joins in `convert_glatos_to_att` happen when the algorithm:

1. creates tag metadata by taking the unique tags in detectionObj, getting common names from WORMS, and joins them together
2. builds on the tag metadata by joining with release records
3. joins detectionObj with receiverObj to match detections with corresponding receiver

Could the common names in detectionObj match with multiple scientific names, returned by WORMS _a la_ #237? Depending on how many receivers you have it could inflate things.

**Comment by mhpob** [comment]
Darn. No idea what the multi-match is here without looking at the data. Have you tried going line-by-line through the function's code? I'm really going to be surprised if this is being tripped up by "only" 6m rows.

**Comment by mhpob** [comment]
It looks like this is doing a VERY GREEDY join, matching every detection to every deployment record, then filtering to only detections at the right station, while the receiver was deployed.

I moved to `dplyr::left_join(detections, receivers, by = dplyr::join_by(between(detection_time, deploy_time, recover_time)))`, but I'm not getting an identical object.

A non-equi join may be the path forward on this, but I need to test that output and get it to match.
